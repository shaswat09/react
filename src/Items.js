import React from 'react';
import PropTypes from 'prop-types';

/**
 * Render a list of items
 *
 * @param {Object} props - List of items
 */
function Items(props) {
  const { items = [] } = props;

  if (!items.length) {
    // No Items on the list, render an empty message
    return <span>No items in list</span>;
  }

  if (items.length === 1) {
    // One Item in the list, render a span
    return <span>{items[0]}</span>;
  }

  // Multiple items on the list, render a list
  return (
    <div>
        <div id="lol">
      <ul>
        {items.map(item => <li key={item}>{item}</li>)}
      </ul>
      </div>
    </div>
  );
}

Items.propTypes = {
  items: PropTypes.array,
};

Items.defaultProps = {
  items: [],
};

export default Items;