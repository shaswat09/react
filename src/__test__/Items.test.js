import React from 'react';
import renderer  from 'react-test-renderer';
import Items from '../Items';

describe("Items",()=>{
    it("default",()=>{

      let tree=  renderer.create(<Items />);
      expect(tree.toJSON()).toMatchSnapshot();
    })

    it("render correctly when single item",()=>{

        const items=['abc'];
   let tree=  renderer.create(<Items  items={items} />);
        expect(tree.toJSON()).toMatchSnapshot();
    })

    it("render correctly when multiple items",()=>{

        const items=['abc','xyz','pqr'];
     let tree=   renderer.create(<Items items={items}/>);
expect(tree.toJSON()).toMatchSnapshot();
    })

})