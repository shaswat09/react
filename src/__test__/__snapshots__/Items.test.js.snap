// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items default 1`] = `
<span>
  No items in list
</span>
`;

exports[`Items render correctly when multiple items 1`] = `
<div>
  <div
    id="lol"
  >
    <ul>
      <li>
        abc
      </li>
      <li>
        xyz
      </li>
      <li>
        pqr
      </li>
    </ul>
  </div>
</div>
`;

exports[`Items render correctly when single item 1`] = `
<span>
  abc
</span>
`;
